kyaml/kio/byteio_reader.go

```diff
@@ -101,6 +102,10 @@ type ByteReader struct {
 	// WrappingKind is set by Read(), and is the kind of the object that
 	// the read objects were originally wrapped in.
 	WrappingKind string
+
+	// Path indicates file path which is being read
+	// empty if it is being read from stdin
+	Path string
```

Comment:
Can we detect the format from the input -- i.e. if the input is JSON parse the JSON, if the input is YAML parse the YAML -- in fact json is a subset of yaml, why do we need to know if this is json at all?